cppForSwig/Makefile

```diff
@@ -40,23 +42,33 @@ else
 PYTHON_INCLUDES=$(shell $(PYVER)-config --includes )
 endif
 
-CPPFLAGS += $(ARMORY_CPPFLAGS) -Icryptopp -Imdb -DUSE_CRYPTOPP -D__STDC_LIMIT_MACROS
-LDLIBS += -lpthread -Lmdb
-SWIG_OPTS    += -c++ -python -threads
-
-SWIG_INC     += 
-
-CXXCPP += $(CPPFLAGS) $(PYTHON_INCLUDES) -std=c++11
+# "-Wl,--no-as-needed" used for Ubuntu 12.04 due to rt library linking issues.
+# http://stackoverflow.com/questions/17150075/undefined-reference-to-clock-gettime-although-lrt-is-given
+# has more info. Require only for static linking. When 12.04 goes, remove this
+# unless it's discovered that OS X needs it.
+ifdef STATIC_LINK
+LDLIBS += -Wl,--no-as-needed
```

Comment:
@droark - Hope you are feeling better. I think the problem with using --no-as-needed is that it isn't limited to librt. So if there are other libraries listed in LDLIBS that aren't required, then they will be linked even though they aren't required. Since Armory doesn't list extraneous libraries in LDLIBS, I don't think there is any practical difference between the --no-as-needed method and the force librt link method. Using the call to force librt to link has the advantage of being targeted to fix the issue by just forcing librt to be linked. I think it's just a matter of limiting the scope of what our fix touches so that we can prevent unintended side effects.

I don't understand the comment about the call only touching 64-bit builds. Don't we need some work around like these ones to get builds working on 12.04, even for 32-bit builds?